examples/lock-app/mbed/README.md

```diff
@@ -0,0 +1,175 @@
+has the commissioner role.
+
+To start the rendezvous, the controller must get the commissioning information
+from the CHIP device. The data payload is encoded within a QR code, printed to
+the UART console.
+
+<a name="wifi_provisioning"></a>
+## Wi-Fi provisioning
+
+Last part of the rendezvous procedure, the provisioning operation involves
+sending the Wi-Fi network credentials from the CHIP controller to the CHIP
+device. As a result, device is able to join the Wi-Fi network and communicate
+with other CHIP devices in the network.
+
+<a name="build_and_flash"></a>
+# Building and Flashing Example Application
+
+<a name="environment_setup"></a>
+## Development environment setup    
+
+Building the example application requires the use of **ARM Mbed-OS** sources 
+and **arm-none-gnu-eabi** toolchain. The OpenOCD package is used for 
+flashing purpose.
+<br>
+Some additional packages may be needed, depending on selected build target 
+and its requirements.
+
+> **The VSCode devcontainer has these components pre-installed, so using 
+the VSCode devcontainer is recommended way to interact with ARM Mbed-OS 
+port of the CHIP Project.**    
+>    
+> **Please read this [README.md](../../..//docs/VSCODE_DEVELOPMENT.md) for 
+more information about using VSCode in container.**
+
+To initialize development environment, download all registered sub-modules 
+and activate development environment:
+```
+$ source ./scripts/bootstrap.sh    
+$ source ./scripts/activate.sh
+```
+If packages are already installed then simply just activate development 
+environment:
+```
+$ source ./scripts/activate.sh
+```
+<a name="building"></a>
+## Building
+
+The Lock-app application can be built in the same way as any other CHIP 
+example ported to mbed-os platform.
+<br>
+* **by using generic vscode task**:  
+```
+Terminal => Run Task... => Run Mbed application => build => (example name) => (board name) => (build profile)`    
+```  
+<br>
+
+* **by calling explicitly building script:**
+```
+${HOME}/scripts/examples/mbed_example.sh -c build -a <example name> -b <board name> -p <build profile>
+```
+Both approaches are limited only to supported evaluation boards which are 
+listed in [Supported devices](#supported_devices) paragraph. 
+<br>When using building script, it is possible expand its list of acceptable 
+targets which may be useful for rapid testing a new mbed-targets.
+<br><br>For more detailed description about porting mbed targets into the 
+CHIP project, please read **[LINK HERE]**
+
+<a name="flashing"></a>
+## Flashing
+
+The Lock-app application can be flashed in the same way as any other CHIP 
+example ported to mbed-os platform.
+<br>
+* **by using generic vscode task**: 
+```
+Terminal => Run Task... -> Run Mbed application => flash => (example name) => (board name) => (build profile)
+```
+<br>
+
+* **by calling explicitly building script:**
+```
+${HOME}/scripts/examples/mbed_example.sh -c flash -a <example name> -b <board name> -p <build profile>
```

Comment:
The part with `${HOME}` can be confused with the user's home directory. It would be better to mention the root of chip.